kodilla-spring/src/main/java/com/kodilla/spring/calculator/Calculator.java

```diff
@@ -0,0 +1,48 @@
+package com.kodilla.spring.calculator;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class Calculator {
+    @Autowired
+    private Display display;
+    @Autowired
+    private DivideByZero divideByZero;
+
+    public double add(double a, double b) {
+        double result = a + b;
+
+        display.displayValue(result);
+        return result;
+    }
+
+    public double sub(double a, double b) {
+        double result = a - b;
+
+        display.displayValue(result);
+        return result;
+    }
+
+    public double mul(double a, double b) {
+        double result = a * b;
+
+        display.displayValue(result);
+        return result;
+    }
+
+    public double div(double a, double b) {
+        double result;
+
+        if ( b != 0) {
+            result = a / b;
+            display.displayValue(result);
+        } else {
+            result = Double.NaN;
```

Comment:
throw new IllegalArgumentException("Dividinv by zero!");

Reply:
albo jeśli tak chcesz, to klasa Display może mieć metodę, która wypisze błąd, np. `public void printAttemptToDivideByZero() { ... }`